vunit/rivierapro_interface.py

```diff
@@ -231,13 +231,7 @@ def _create_load_function(self,
 
         vsim_flags = ["-dataset {%s}" % fix_path(join(output_path, "dataset.asdb")),
                       pli_str,
-                      set_generic_str,
-                      "-lib",
-                      config.library_name,
-                      config.entity_name]
-
-        if config.architecture_name is not None:
-            vsim_flags.append(config.architecture_name)
+                      set_generic_str]
```

Comment:
What happens when the set generic string is separated from the top level? Have you tried this on a test bench which have generics set from Python?

Reply:
I tried shuffling the arguments around with rand.shuffle() (as a part of diagnosing the coverage issues) and never noticed any issues with the re-ordering (except getting coverage data or not depending on the order of the testbench and glbl).

If you think it helps readability I can move the set_generic_str command down too.